tcms/rpc/api/testexecution.py

```diff
@@ -114,6 +114,9 @@ def update(execution_id, values):
     if values.get('case_text_version') == 'latest':
         values['case_text_version'] = test_execution.case.history.latest().history_id
 
+    if values.get('status'):
```

Comment:
status is a foreign key and I think status_id is also an accepted value for the form field.

Reply:
I wrote this:
```python
    def test_update_with_status_id_changes_tested_by(self):
        self._test_update_status_with_key_changes_tested_by('status_id')

    def test_update_with_status_changes_tested_by(self):
        self._test_update_status_with_key_changes_tested_by('status')

    def _test_update_status_with_key_changes_tested_by(self, status_key):
        execution = TestExecutionFactory(
            tested_by=self.user
        )

        self.rpc_client.TestExecution.update(execution.pk, {status_key: self.status_positive.pk})
        execution.refresh_from_db()

        self.assertEqual(execution.tested_by, self.api_user)
        self.assertEqual(execution.status, self.status_positive)
```
and `test_update_with_status_id_changes_tested_by` failed with:
```
======================================================================
FAIL: test_update_with_status_id_changes_tested_by (tcms.rpc.tests.test_testexecution.TestExecutionUpdate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/anton_sankov/Desktop/reno/git/kiwitcms/Kiwi/tcms/rpc/tests/test_testexecution.py", line 385, in test_update_with_status_id_changes_tested_by
    self._test_update_status_with_key_changes_tested_by('status_id')
  File "/Users/anton_sankov/Desktop/reno/git/kiwitcms/Kiwi/tcms/rpc/tests/test_testexecution.py", line 399, in _test_update_status_with_key_changes_tested_by
    self.assertEqual(execution.status, self.status_positive)
AssertionError: <TestExecutionStatus: IDLE> != <TestExecutionStatus: WAIVED>
```
so I guess `status_id` does not work here

Reply:
I've tried it out and you are correct. For FK fields they don't accept names with `_id` suffix even though that doesn't make the form invalid. I suspect this is a side effect from using ModelForm but I didn't dig deep enough.